modules/lib/lib-scheduler/src/main/resources/lib/xp/scheduler.ts

```diff
@@ -0,0 +1,229 @@
+
+export interface CreateScheduledJobParams {
+    name: string;
+    description: string;
+    descriptor: string;
+    config: Record<string, Object>;
+    schedule: Schedule;
+    user: string;
+    enabled: boolean;
+}
+
+export interface OnTimeSchedule {
+    value: string;
+    type: ScheduleType;
+}
+
+export interface CronSchedule
+    extends OnTimeSchedule {
+    timeZone: string;
+}
+
+export interface ScheduleJob {
+    name: string;
+    descriptor: string;
+    description: string;
+    enabled: boolean;
+    config?: Record<string, Object> | null;
+    user?: string | null;
+    creator?: string | null;
+    modifier?: string | null;
+    createdTime?: string | null;
+    modifiedTime?: string | null;
+    lastRun?: string | null;
+    lastTaskId?: string | null;
+    schedule: OnTimeSchedule | CronSchedule;
+}
+
+interface CreateScheduledJobHandler {
+    setName(value: string): void;
+
+    setSchedule(value: Schedule): void;
+
+    setDescriptor(value: string): void;
+
+    setEnabled(value: boolean): void;
+
+    setDescription(value?: string | null): void;
+
+    setConfig(value?: object | null): void;
+
+    setUser(value?: string | null): void;
+
+    execute(): ScheduleJob;
+}
+
+/**
+ * Creates a job to be scheduled.
+ *
+ * @example-ref examples/scheduler/create.js
+ *
+ * @param {object} params JSON with the parameters.
+ * @param {string} params.name unique job name.
+ * @param {string} params.description job description.
+ * @param {string} params.descriptor descriptor of the task to be scheduled.
+ * @param {object} params.config config of the task to be scheduled.
+ * @param {object} params.schedule task time run config.
+ * @param {string} params.schedule.value schedule value according to it's type.
+ * @param {string} params.schedule.type schedule type (CRON | ONE_TIME).
+ * @param {string} params.schedule.timezone time zone of cron scheduling.
+ * @param {string} params.user key of the user that submitted the task.
+ * @param {boolean} params.enabled job is active or not.
+ */
+export function create(params: CreateScheduledJobParams): ScheduleJob {
+    checkRequired(params, 'name');
+    checkRequired(params, 'schedule');
+    checkRequired(params, 'descriptor');
+    checkRequired(params, 'enabled');
+
+    const bean = __.newBean<CreateScheduledJobHandler>('com.enonic.xp.lib.scheduler.CreateScheduledJobHandler');
+
+    bean.setName(params.name);
+    bean.setSchedule(params.schedule);
+    bean.setDescriptor(params.descriptor);
+    bean.setEnabled(params.enabled);
+    bean.setDescription(__.nullOrValue(params.description));
+    bean.setConfig(__.toScriptValue(params.config));
+    bean.setUser(__.nullOrValue(params.user));
+
+    return __.toNativeObject(bean.execute());
+}
+
+export interface ModifyScheduledJobParams {
+    name: string;
+    editor: (schedule: ScheduleJob) => ScheduleJob;
+}
+
+interface ModifyScheduledJobHandler {
+    setName(value: string): void;
+
+    setEditor(value: Object): void;
```

Comment:
Look to the previous PR and see how we can type and replace this with `EditorFn<T>`

Reply:
Thank you for your suggestion!